test/libsolidity/syntaxTests/nameAndTypeResolution/313_fixed_type_size_capabilities.sol

```diff
@@ -10,4 +10,5 @@ contract test {
     }
 }
 // ----
-// Warning: (20-707): Function state mutability can be restricted to pure
+// TypeError: (153-250): Type rational_const 9208...(70 digits omitted)...7637 / 1000...(71 digits omitted)...0000 is not implicitly convertible to expected type ufixed256x77. Try converting to type ufixed256x77 or use an explicit conversion.
```

Comment:
The error message does not really fit anymore. Can you explain why this is not convertible anymore?

Reply:
I've made a patch to this ([here](https://github.com/0yi0/solidity/commit/f7c02e7e073c1f9c107e437146255f35f0cca0b2), haven't picked it into this PR yet).

But changing `RationalNumberType::fixedPointType()` will also cause the behavior of explicit conversion to change, since it's using the same method mentioned in #5551, too.

```
bool RationalNumberType::isExplicitlyConvertibleTo(Type const& _convertTo) const
{
    if (isImplicitlyConvertibleTo(_convertTo))
        return true;
    else if (_convertTo.category() != Category::FixedBytes)
    {
        TypePointer mobType = mobileType();
        return (mobType && mobType->isExplicitlyConvertibleTo(_convertTo));
    }
    else
        return false;
}
```

And this will cause another test to fail (unexpectedly).

```
error in "fixed_point_division": Test expectation mismatch.
Expected result:
  TypeError: (32-49): Type rational_const 10000000000000000 / 27 is not implicitly convertible to expected type uint256.
Obtained result:
  TypeError: (32-49): Type rational_const 10000000000000000 / 27 is not implicitly convertible to expected type uint256.
  TypeError: (72-97): Explicit type conversion not allowed from "rational_const 10000000000000000 / 27" to "ufixed128x18".
```

I'm not sure what literals should NOT be explicitly convertible to a fixed point numbers,
so I have no idea how to fix this test.

Reply:
In my opinion, a literal value that is convertible to `ufixedMxN` should be in the range of `[0, (pow(2, M)-1)/pow(10,N)]`. So `2.56` should not fit into `ufixed8x1`.

But currently integers do not work in this way. The below code compiles without any errors or warnings.

```
contract C { 
    function explicitConvertible() public pure {
        uint8 a = uint8(-1);
        uint8 b = uint8(512);
        a; b;
    }   
}
```

I think the behavior is not clearly documented so I'm not sure whether it's a bug or a feature.

Reply:
I agree with the first sentence. The example you provide performs explicit conversions, which is a different thing.

Reply:
Ok, looking at the code again, I think I understand why we get this warning now, and it does make sense. The number in question cannot be *implicitly* converted to `fixed256x77`, but it can be *explicitly* converted. Could you change the error message to make that clearer?

Reply:
Sounds good to me.

I've updated the expected messages in the tests. I'll update the error message in the next commit.